src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,93 @@
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedModal: Todo
+  onSelectedModal: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedModal,
+  onSelectedModal,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoding, setIsLoading] = useState(true);
+  const [hasLoadingModalError, setHasLoadingModalError] = useState(false);
+
+  useEffect(() => {
+    getUser(selectedModal.userId)
+      .then(setUser)
+      .catch(() => setHasLoadingModalError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const { id, completed } = selectedModal;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {isLoding && (
         <Loader />
-      ) : (
+      )}
+
+      {!isLoding && (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
+              aria-label="Mute volume"
               className="delete"
               data-cy="modal-close"
+              onClick={() => onSelectedModal(null)}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {selectedModal.title}
             </p>
 
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+            {hasLoadingModalError
+              ? (
+                <div className="notification is-danger">
+                  An error occured when loading data!
+                </div>
+              )
+              : (
+                <p className="block" data-cy="modal-user">
+                  <strong className={classNames({
+                    'has-text-success': completed,
+                    'has-text-danger': !completed,
+                  })}
+                  >
+                    {selectedModal.completed
+                      ? 'Done'
+                      : 'Planned'}
+                  </strong>
 
-              {' by '}
+                  {' by '}
 
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+                  <a href={`mailto:${user?.email}`}>
+                    {user?.name}
+                  </a>
```

Comment:
when there is no name, then there is no sense to render an empty link. Try to avoid redundant html generation.
```
 {user && 
     <> 
         by 
         <a href={`mailto:${user.email}`}>
            {user.name}
         </a>
     </>
 }
```